src/common/utils/mod.rs

```diff
@@ -14,6 +14,7 @@
 // along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 pub mod auth;
+mod auth_tests;
```

Comment:
Eventually I think it would make sense to move the `AuthExtractor` and its unit tests all into the same submodule (inside the `auth` module). However, since this PR is already very large I want it to focus exclusively on adding new tests for `AuthExtractor`.